### PR TITLE
[TACHYON-574] Move empty-body for-loop in tachyon.examples.Performance to commonUtils

### PR DESCRIPTION
**Pull request by zhljen**
https://tachyon.atlassian.net/browse/TACHYON-574

### CODE REVIEW

**Comment by calvinjia** [comment]
Jenkins, add to whitelist.

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
I'm not sure if the sleep achieves the same effect of warm up as the empty for loop. @apc999

**Comment by zhljen** [comment]
maybe we can measure the time of the for loop to verify the time?

**Comment by haoyuan** [comment]
The empty loop is to warmup the JVM.

**Comment by zhljen** [comment]
Ok. Looks like we need the for loop. If this change is not needed, should
we close the ticket as well?

**Comment by apc999** [comment]
Hi @zhljen , thanks for submitting this patch! I didn't know this warmup until saw @haoyuan's reply.
Here is my suggestion for this PR: 

(1) create some method called `warmup` or`warmupBenchmark` or something similar in `tachyon.util.CommonUtils`

(2) replace the for-loop in this JIRA with this newly created util method.

In this way, we could have a better indicator what this piece of code is doing and also provide a consistent way for all other benchmarks (coming very soon).

Does it sound good to you? Any other suggestions are welcome

**Comment by zhljen** [comment]
Sounds reasonable to me. The approach will help people understand the code,
and make the code cleaner.
Zhen

**Comment by yuzhu** [comment]
Just happen to stumble on this,  is an empty loop like this sufficient to warm up a JVM? 

I think we want the JVM to JIT the code path that we are testing before we time them. In that case, should we send some write/read requests in the loop to warm up the code path of reading and writing in tachyon to get a more accurate performance result? (could be a separate JIRA though)

**Comment by apc999** [comment]
@yuzhu , totally agree. I checked some discussion on stackoverflow, it seems using an empty loop as JVM warmup has no guarantee at all. So that 's why we want in this PR to separate this logic out and you are welcome to create new JIRA to explore what's the best practice for this purpose.

File: common/src/main/java/tachyon/util/CommonUtils.java [hunk 1]
```diff
@@ -64,6 +64,10 @@ public static void sleepMs(Logger logger, long timeMs, boolean shouldInterrupt)
     }
   }
 
+  public static void warmUpJVM() {
+    for (int k = 0; k < 10000000; k ++) {
+    }
+  }
```
Comment (apc999) [inline common/src/main/java/tachyon/util/CommonUtils.java]: A minor format issue: please leave one extra line after this function before the javadoc of the next one.

File: common/src/main/java/tachyon/util/CommonUtils.java [hunk 2]
```diff
@@ -64,6 +64,10 @@ public static void sleepMs(Logger logger, long timeMs, boolean shouldInterrupt)
     }
   }
 
+  public static void warmUpJVM() {
```
Comment (apc999) [inline common/src/main/java/tachyon/util/CommonUtils.java]: Please javadoc what this function intends to do, and mark that there might be no guarantee.

**Comment by apc999** [comment]
@zhljen I left two minor comments. Thanks for the update.

**Comment by zhljen** [comment]
Researched a bit on the JVM warmup, which is quite involved and situation
dependent. Given this, I will rename the function to warmupLoop or
something similar to reflect its functionality, instead of warmupJVM. Let
me know if you have any comments.

**Comment by apc999** [comment]
@zhljen your proposal sounds reasonable to me. Just remember to make similar statement in the javadoc of this util function.

**Comment by apc999** [comment]
+1 LGTM

File: common/src/main/java/tachyon/util/CommonUtils.java [hunk 3]
```diff
@@ -64,6 +64,11 @@ public static void sleepMs(Logger logger, long timeMs, boolean shouldInterrupt)
     }
   }
 
+  public static void warmUpLoop() {
```
Comment (calvinjia) [inline common/src/main/java/tachyon/util/CommonUtils.java]: Could you add the java doc to this function please?

Comment (zhljen) [inline common/src/main/java/tachyon/util/CommonUtils.java]: Added just now.

**Comment by calvinjia** [comment]
+1 LGTM

**Pull request merged**